File: OpenMPvalidation/SRC/C/check_omp_schedule_main.c
#include <stdio.h>
#include <assert.h>

#include <omp.h>
#include "omp_testsuite.h"


int main(int argc,char** argv){
  
  printf(" Has _OPENMP macro ...");
  if (check_has_openmp() ){
    printf(" passed\n");
  } else {
    printf(" failed\n");
  }
  
#ifndef USE_ASSURE
  printf(" omp_get_num_threads ...");
  if (check_omp_get_num_threads() ){
    printf(" passed\n");
  } else {
    printf(" failed\n");
  }
#endif

  printf("omp in schedule...");
if(check_for_schedule())
{
printf("passed\n");
}
else 
{printf("failed\n");}
return 0;
}

File: OpenMPvalidation/SRC/F90/check_omp_schedule_main.f
	program main
        implicit none
        integer temp
        integer check_has_openmp
        integer chk_omp_get_num_threads
        integer check_do_schedule
!	include "omp_testsuite_include.f"
	print *, ' Has OpenMP macro ...'
	if (check_has_openmp().eq. 1) then
	  print *, ' passed'	
	else
	  print *, ' failed'
	end if

	print *, ' omp_get_num_threads ...'
        if (chk_omp_get_num_threads().eq. 1) then
          print *, ' passed'
        else
          print *, ' failed'
        end if

	print *, ' omp in schedule...'
        if ( check_do_schedule().eq. 1) then
          print *, ' passed'
        else
          print *, ' failed'
        end if
	end
